katdal/test/test_lazy_indexer.py

```diff
@@ -123,77 +157,34 @@ def numpy_oindex_lite(x, keep):
-
-    def _test_with(self, indices, indices_without_ellipsis=None):
-        """An even more compact version of numpy_oindex..."""
-        outer = numpy_oindex(self.data, indices)
-        if indices_without_ellipsis is None:
-            indices_without_ellipsis = indices
-        expected = numpy_oindex_lite(self.data, indices_without_ellipsis)
-        np.testing.assert_array_equal(outer, expected)
-
-    def test_outer_indexing(self):
-        self._test_with(())
-        self._test_with(2)
-        self._test_with((2, 3, 4, 5))
-        # ellipsis
-        self._test_with(np.s_[[0], ...], np.s_[[0], :, :, :])
-        self._test_with(np.s_[:, [0], ...], np.s_[:, [0], :, :])
-        # list of ints
-        self._test_with(np.s_[:, [0], [0], :])
-        self._test_with(np.s_[:, [0], :, [0]])
-        self._test_with(np.s_[:, [0], [0, 1], :])
-        self._test_with(np.s_[:, [0], :, [0, 1]])
-        # booleans
-        pick_one = np.zeros(10, dtype=np.bool_)
-        pick_one[6] = True
-        self._test_with(np.s_[:, [True, False] * 5, pick_one, :])
-        self._test_with(np.s_[:, [False, True] * 5, :, pick_one])
-        # slices
-        self._test_with(np.s_[0:2, 2:4, 4:6, 6:8])
-        self._test_with(np.s_[-8:-6, -4:-2, slice(3, 10, 2), -2:])
-        # single ints
-        self._test_with(np.s_[:, [0], 0, :])
-        self._test_with(np.s_[:, [0], :, 0])
-        self._test_with(np.s_[:, [0], -1, :])
-        self._test_with(np.s_[:, [0], :, -1])
-        # newaxis
-        self._test_with(np.s_[np.newaxis, :, [0], :, 0])
-        self._test_with(np.s_[:, [0], np.newaxis, 0, :])
-        # the lot
-        self._test_with(np.s_[..., 2:5, [4, 6], np.newaxis, 0],
-                        np.s_[:, 2:5, [4, 6], np.newaxis, 0])
-
-
 class TestDaskGetitem(object):
     """Test the :func:`~katdal.lazy_indexer._dask_getitem` function."""
     def setup(self):
         shape = (10, 10, 10, 10)
         self.data = np.arange(np.product(shape)).reshape(shape)
         self.data_dask = da.from_array(self.data, chunks=(2, 5, 2, 5))
 
-    def _test_with(self, indices):
+    def _test_with(self, indices, indices_without_ellipsis=None):
         npy = numpy_oindex(self.data, indices)
+        if indices_without_ellipsis is None:
+            indices_without_ellipsis = indices
+        npy_lite = numpy_oindex_lite(self.data, indices_without_ellipsis)
         getitem = _dask_getitem(self.data_dask, indices).compute()
+        np.testing.assert_array_equal(npy, npy_lite)
         np.testing.assert_array_equal(getitem, npy)
 
     def test_outer_indexing(self):
         self._test_with(())
         self._test_with(2)
         self._test_with((2, 3, 4, 5))
         # ellipsis
-        self._test_with(np.s_[[0], ...])
-        self._test_with(np.s_[:, [0], ...])
+        self._test_with(np.s_[[0], ...], np.s_[[0], :, :, :])
+        self._test_with(np.s_[:, [0], ...], np.s_[:, [0], :, :])
```

Comment:
Maybe have a test where `...` is in the middle rather than at one end.

Reply:
Done.